src/main.rs

```diff
@@ -44,8 +45,24 @@ pub fn main() {
 
     match builder_from_flags(bind_args.into_iter()) {
         Ok((builder, output)) => {
-            let mut bindings = builder.generate()
-                .expect("Unable to generate bindings");
+
+            let builder_result =
+                panic::catch_unwind(||
```

Comment:
Is this what you mean?

```rust
panic::catch_unwind(AssertUnwindSafe(|| -> io::Result<()> {
    let mut bindings = builder.generate().expect("Unable to generate bindings");
    try!(bindings.write(output));
    try!(bindings.write_dummy_uses());
    Ok(())
}));
```
I don't immediately see how it can be done without using `UnwindSafe` or `AssertUnwindSafe` because `catch_unwind` expects something with the `UnwindSafe` trait. Could you clarify a bit more?